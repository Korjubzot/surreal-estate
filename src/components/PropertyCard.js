/* eslint-disable import/no-extraneous-dependencies */
import React from "react";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import {
  faSink,
  faBed,
  faSterlingSign,
  faEnvelope,
} from "@fortawesome/free-solid-svg-icons";

const PropertyCard = (props) => {
  const { title, city, type, bedrooms, bathrooms, price, email } = props;

  return (
    <div className="property-card">
      <h4 className="title-property-card">{title}</h4>
      <h5 className="type-city">
        {type} - {city}
      </h5>
      <h5 className="bathrooms">
        <FontAwesomeIcon icon={faBed} /> {bathrooms}
      </h5>
      <h5 className="bedrooms">
        <FontAwesomeIcon icon={faSink} /> {bedrooms}
      </h5>
      <h5 className="price">
        <FontAwesomeIcon icon={faSterlingSign} /> {price}
      </h5>
      <h5 className="email">
        <FontAwesomeIcon icon={faEnvelope} />
        &nbsp;
        <a href={`mailto:${email}`}>Contact</a>
      </h5>
    </div>
  );
};

export default PropertyCard;
