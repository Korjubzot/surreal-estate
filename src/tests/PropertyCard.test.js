import React from "react";
import { render, screen } from "@testing-library/react";
import PropertyCard from "../components/PropertyCard";

describe("Property Card", () => {
  test("renders the PropertyCard component", () => {
    const validProps = {
      title: "2 bed city center",
      city: "Manchester",
      type: "Flat",
      bedrooms: 2,
      bathrooms: 2,
      price: 500000,
      email: "john.smith@gmail.com",
    };
    const { asFragment } = render(<PropertyCard props={validProps} />);

    expect(asFragment()).toMatchSnapshot();
  });
});
