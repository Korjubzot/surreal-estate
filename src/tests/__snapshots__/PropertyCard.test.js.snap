// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Property Card renders the PropertyCard component 1`] = `
<DocumentFragment>
  <div
    class="property-card"
  >
    <h4
      class="title-property-card"
    />
    <h5
      class="type-city"
    >
       - 
    </h5>
    <h5
      class="bathrooms"
    >
       bathrooms
    </h5>
    <h5
      class="bedrooms"
    >
       bedrooms
    </h5>
    <h5
      class="price"
    >
      Â£
    </h5>
    <h5
      class="email"
    >
      <a
        href="mailto:undefined"
      >
        Contact
      </a>
    </h5>
  </div>
</DocumentFragment>
`;
